python/idsse_common/idsse/common/protocol_utils.py

```diff
@@ -0,0 +1,173 @@
+"""Base class for http and awc data access"""
+# -------------------------------------------------------------------------------
+# Created on Tue Dec 3 2024
+#
+# Copyright (c) 2023 Colorado State University. All rights reserved.             (1)
+# Copyright (c) 2023 Regents of the University of Colorado. All rights reserved. (2)
+#
+# Contributors:
+#     Paul Hamer (1)
+#
+# -------------------------------------------------------------------------------
+import fnmatch
+import os
+
+from abc import abstractmethod, ABC
+from collections.abc import Sequence
+from datetime import datetime, timedelta, UTC
+
+from .path_builder import PathBuilder
+from .utils import TimeDelta, datetime_gen
+
+class ProtocolUtils(ABC):
+    """Base Class - interface for DAS data discovery"""
+
+    def __init__(self,
+                 basedir: str,
+                 subdir: str,
+                 file_base: str,
+                 file_ext: str) -> None:
+        self.path_builder = PathBuilder(basedir, subdir, file_base, file_ext)
+
+    # pylint: disable=invalid-name
+    @abstractmethod
+    def ls(self, path: str, prepend_path: bool = True) -> Sequence[str]:
+        """Execute a 'ls' on the specified path
+
+        Args:
+            path (str): path
+            prepend_path (bool): Add to the filename
+
+        Returns:
+            Sequence[str]: The results sent to stdout from executing a 'ls' on passed path
+        """
+
```

Comment:
I think it makes sense to add a 'cd' abstract method, since both aws and http implement them.